src/main/java/com/conveyal/r5/profile/RaptorState.java

```diff
@@ -337,4 +340,28 @@ public boolean stopWasUpdated (int stop, boolean withinMinute) {
         }
     }
 
+    /**
+     * Checks whether boarding a at one stop allows for shorter access/transfer legs than boarding at another stop.
+     */
+    public boolean shorterAccessOrTransferLeg(int here, int there) {
+
+        if (here == there) return false;
+
+        if (this.previous == null) {
+            // In the first transit round (when the pre-transit state is the input state), arriving at this stop before
+            // arriving at the upstream stop implies shorter access time.
+            return this.bestTimes[here] < this.bestTimes[there];
+        } else {
+            // In subsequent transit rounds, we want to compare the length of the transfer legs.
+            // We cannot directly compare arrival times or waiting times at the stops, because the transfer legs may
+            // have started at different times.
+            return transferTime(here) < transferTime(there);
+        }
+    }
+
+    private int transferTime(int stop) {
```

Comment:
Continuing the Javadoc expansion from shorterAccessOrTransferLeg, this method should state that it returns the length of time spent walking (or using some other street mode) within this round, to achieve the optimal route to the given stop in this round.